composer/core/evaluator.py

```diff
@@ -59,6 +61,46 @@ def should_eval(state: State, event: Event):
             last_batch_seen = state.timestamp.batch
             return True
 
+        if eval_interval.unit == TimeUnit.DURATION:
+            if state.max_duration is None:
+                raise ValueError(f'`max_duration` is a required argument and must be specified when constructing the '
+                                 f'`Trainer()` or `duration` when calling `Trainer.fit()`')
```

Comment:
Since we check this in `Trainer`, it's fine to skip this and not have to raise a ValueError. If it's because of pyright typing, `assert state.max_duration != None` is fine.

Reply:
Done in https://github.com/mosaicml/composer/pull/1523/commits/7fb861f1fd1c6b4461132c9092ec826f48a00071.